nav2_behavior_tree/test/plugins/action/test_controller_cancel_node.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2022 Neobotix GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <memory>
+#include <set>
+#include <string>
+
+#include "behaviortree_cpp_v3/bt_factory.h"
+
+#include "../../test_action_server.hpp"
+#include "nav2_behavior_tree/plugins/action/controller_cancel_node.hpp"
+#include "lifecycle_msgs/srv/change_state.hpp"
+
+class CancelControllerServer : public TestActionServer<nav2_msgs::action::FollowPath>
+{
+public:
+  CancelControllerServer()
+  : TestActionServer("follow_path")
+  {}
+
+protected:
+  void execute(
+    const typename std::shared_ptr<rclcpp_action::ServerGoalHandle<nav2_msgs::action::FollowPath>>
+    goal_handle)
+  {
+    while (!goal_handle->is_canceling()) {
+      // waiting here until goal cancels
```

Comment:
Add a sleep here, a while empty condition will explode and consume all available resources. A sleep for 10ms will give the scheduler time to do other tasks